internal/controller/operator/factory/build/child_object.go

```diff
@@ -0,0 +1,129 @@
+package build
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	vmv1beta1 "github.com/VictoriaMetrics/operator/api/operator/v1beta1"
+	"github.com/VictoriaMetrics/operator/internal/controller/operator/factory/logger"
+)
+
+var badObjectsTotal = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "operator_bad_objects_count",
+		Help: "Number of incorrect objects by controller",
+	},
+	[]string{"crd", "object_namespace"},
+)
+
+func init() {
+	metrics.Registry.MustRegister(badObjectsTotal)
+}
+
+type scrapeObjectWithStatus interface {
+	client.Object
+	GetStatusMetadata() *vmv1beta1.StatusMetadata
+	AsKey(bool) string
+}
+
+type ChildObjects[T scrapeObjectWithStatus] struct {
```

Comment:
There is no any kind of documentation to the public exported methods.

Reply:
added comments